docs/developer-guide/working-with-plugins.md

```diff
@@ -18,6 +18,21 @@ module.exports = {
 };
 ```
 
+### Preprocessors in Plugins
+
+You can also create plugins that would tell ESLint how to process files other then JavaScript. In order to create a
+preprocessor, object that is exported from your module has to conform to the following interface:
+
+```js
+{
+    preprocess: function(string) { return [string] } //Takes text of the file and returns array of texts to be processed
+    postprocess: function([[Message]]) { return [Message] //Takes array of array of error messages, one for each text
```

Comment:
@nzakas Any suggestions on the format of the export object from the plugin? I want to make `preprocessors` just an array, but all other properties are objects. I have no idea what to use as a key for preprocessors object. Extension wouldn't work, since preprocessors should be able to handle multiple extensions (html, htm, etc.).